src/icecast/audio/PlaylistAudioSource.cpp

```diff
@@ -0,0 +1,178 @@
+#include "PlaylistAudioSource.h"
+
+#include "AudioSourceType.h"
+#include "AudioMetadataChangedEventArgs.h"
+#include "../config/NoiseStreamerConfig.h"
+#include "../../kernel/audio/encoding/lame/LameAudioEncodingService.h"
+#include "../../kernel/exception/domain/DomainException.h"
+#include "../../kernel/exception/ExceptionMapper.h"
+#include "../../kernel/utils/FileHelper.h"
+
+PlaylistAudioSource::PlaylistAudioSource(
+    ILogService* logSrv,
+    SignalService* sigSrv,
+    ITimeService* timeSrv,
+    AudioEncodingService *encSrv)
+    : AudioSource(logSrv, sigSrv, SOURCE_PLAYLIST),
+    NoiseStreamerPlaylist(logSrv, timeSrv, encSrv),
+    encSrv(encSrv)
+{
+    this->currentMp3File = NULL;
+    this->currentNssItem = NULL;
+}
+
+PlaylistAudioSource::~PlaylistAudioSource()
+{
+    currentPlaylistItemFinished();
+}
+
+void PlaylistAudioSource::initialize(AudioSourceConfig config)
+{
+    initializePlaylist((NoiseStreamerConfig*) &config);
```

Comment:
This cause segmentation fault..
We cannot cast from empty value (`AudioSourceConfig`) into pointer (`NoiseStreamerConfig`). The pointer references to an empty object..